src/bdk_cli.rs

```diff
@@ -112,25 +115,75 @@ fn prepare_home_dir() -> Result<PathBuf, Error> {
         fs::create_dir(&dir).map_err(|e| Error::Generic(e.to_string()))?;
     }
 
-    #[cfg(not(feature = "compact_filters"))]
-    dir.push("database.sled");
+    dir.push(wallet_name);
+
+    if !dir.exists() {
+        info!("Creating wallet directory {}", dir.as_path().display());
+        fs::create_dir(&dir).map_err(|e| Error::Generic(e.to_string()))?;
+    }
 
-    #[cfg(feature = "compact_filters")]
-    dir.push("compact_filters");
     Ok(dir)
 }
 
-fn open_database(wallet_opts: &WalletOpts) -> Result<Tree, Error> {
-    let mut database_path = prepare_home_dir()?;
-    let wallet_name = wallet_opts
-        .wallet
-        .as_deref()
-        .expect("We should always have a wallet name at this point");
-    database_path.push(wallet_name);
-    let database = sled::open(database_path)?;
-    let tree = database.open_tree(&wallet_name)?;
+/// Prepare wallet database directory
+fn prepare_wallet_db_dir(wallet_name: &str) -> Result<PathBuf, Error> {
+    let mut db_dir = prepare_home_wallet_dir(wallet_name)?;
+
+    #[cfg(feature = "key-value-db")]
+    db_dir.push("wallet.sled");
+
+    #[cfg(feature = "sqlite-db")]
+    db_dir.push("wallet.sqlite");
+
+    #[cfg(not(feature = "sqlite-db"))]
+    if !db_dir.exists() {
+        info!("Creating database directory {}", db_dir.as_path().display());
+        fs::create_dir(&db_dir).map_err(|e| Error::Generic(e.to_string()))?;
+    }
+
+    Ok(db_dir)
+}
+
+/// Prepare blockchain data directory (for compact filters)
+#[cfg(feature = "compact_filters")]
+fn prepare_bc_dir(wallet_name: &str) -> Result<PathBuf, Error> {
```

Comment:
Just a question: something like this could also be used for auto regtest node datadir right?

Reply:
Yes! I'd like to put the auto regtest data someplace like:
* `~/.bdk-bitcoin/<wallet_name>/regtest/bitcoind/` and
* `~/.bdk-bitcoin/<wallet_name>/regest/electrsd/`.